packages/tools/lint-package/src/index.ts

```diff
@@ -0,0 +1,46 @@
+import { loadContents, loadJSON, loadJSONC, loadRules } from './helpers.js';
+import { Issue, Rule, Rules, TSConfig } from './types.js';
+
+import type { PackageJson } from '@npmcli/package-json';
+import fg from 'fast-glob';
+import { join } from 'node:path';
+
+const PACKAGE_DIR = process.cwd();
+const PACKAGE_JSON = join(PACKAGE_DIR, 'package.json');
+const TS_CONFIG = join(PACKAGE_DIR, 'tsconfig.json');
+const SEPARATOR = '\n        warning  '; // Emulate eslint-like output
+
+const pkgRules = await loadRules('package-json');
+const tsRules = await loadRules('typescript');
+const eslintRules = await loadRules('eslint');
+const apiExtractorRules = await loadRules('api-extractor');
+
+const pkg = await loadJSON<PackageJson>(PACKAGE_JSON);
+const tsConfig = await loadJSONC<TSConfig>(TS_CONFIG);
+const [eslintConfigPath] = await fg('.eslintrc*', { cwd: PACKAGE_DIR });
+const eslintConfig = await loadContents(eslintConfigPath);
+
```

Comment:
Should we add the required files to the loader of each linter?

So each lint rule exports a function/object with the files it needs? That way this file doesn't need to know about the files the rules need, but it can still remove duplicates.

```ts

type RequiredFiles = Array<{ path: string }>;

type LintRule = {
  files: RequiredFiles;
  rules: Array<{
    ruleName: string;
    rule: (files: RequiredFiles[number] & { getContents: () => string }) => Issue[];
  }>;
};
```

Reply:
In my implementation, simply all file contents are sent to each rule (in `options`). This is a bit coarse, but sometimes we need more than just the relevant file (e.g. when having `config/api-extractor.json` we need `package.json` too). The advantage is that (future) rules can read (and fix) all of them to do their thing.

Having that said, this code is not very pretty. Improving the rules loading mechanism could be a nice follow-up.